packages/files-ui/src/Contexts/FilesContext.tsx

```diff
@@ -507,73 +525,97 @@ const FilesProvider = ({ children }: FilesContextProps) => {
+          // Idea for parallel download https://glebbahmutov.com/blog/run-n-promises-in-parallel/
+          // we need to use a reduce here because forEach doesn't wait for the Promise to resolve
+          await fullStructure.reduce(async (totalDownloaded: Promise<number>, item: FileSystemItemPath, index: number): Promise<number> => {
+            const file = await getFileContent(bucketId, {
+              cid: item.cid,
+              file: item,
+              path: getPathWithFile(item.path, item.name),
+              cancelToken,
+              onDownloadProgress: async (progressEvent) => {
+                const currentFileNumber = index + 1
+                const fileProgress = totalFileNumber > 1 && `${currentFileNumber}/${totalFileNumber}`
+                updateToast(toastId, {
+                  ...toastParams,
+                  title: t`Downloading ${fileProgress} - ${item.name}`,
+                  progress: Math.ceil(
+                    ((await totalDownloaded + progressEvent.loaded) / totalFileSize) * 100
+                  )
+                })
+              }
+            })
+
+            if(file) {
+              const fileArrayBuffer = await file.arrayBuffer()
+              const fullPath = getPathWithFile(item.path, item.name)
+              const relativeFilePath = getRelativePath(fullPath, currentPath)
+              zipList[relativeFilePath] = new Uint8Array(fileArrayBuffer)
             }
-          })
-
-          if(file) {
-            const fileArrayBuffer = await file.arrayBuffer()
-            const fullPath = getPathWithFile(item.path, item.name)
-            const relativeFilePath = getRelativePath(fullPath, currentPath)
-            zipList[relativeFilePath] = new Uint8Array(fileArrayBuffer)
-          }
 
-          return await totalDownloaded + item.size
-        }, Promise.resolve(0))
+            return await totalDownloaded + item.size
+          }, Promise.resolve(0))
 
-        // level 0 means without compression
-        const zipped = zipSync(zipList, { level: 0 })
+          // level 0 means without compression
+          const zipped = zipSync(zipList, { level: 0 })
 
-        if (!zipped) return
+          if (!zipped) return
 
-        const link = document.createElement("a")
-        link.href = URL.createObjectURL(new Blob([zipped]))
-        link.download = "archive.zip"
-        link.click()
-
-        updateToast(toastId, {
-          title: t`Download Complete`,
-          type: "success",
-          progress: undefined
-        }, true)
-
-        URL.revokeObjectURL(link.href)
+          const link = document.createElement("a")
+          link.href = URL.createObjectURL(new Blob([zipped]))
+          link.download = "archive.zip"
+          link.click()
+          setDownloadsInProgress(false)
 
+          updateToast(toastId, {
+            title: t`Download Complete`,
+            type: "success",
+            progress: undefined,
+            onProgressCancel: undefined
```

Comment:
Can we add the `isClosable: true` here?
```suggestion
            onProgressCancel: undefined,
            isClosable: true
```

Reply:
Yes makes sense, after operations are complete!